ENG-3475: Tenant-scoped identity providers (#3995)
FusionAuth/fusionauth-site: astro/src/content/docs/lifecycle/authenticate-users/identity-providers/index.mdx

If you're looking for a provider that is not listed here, review the open features in [GitHub](https://github.com/FusionAuth/fusionauth-issues/issues) and either vote or comment on an existing feature, or open a new feature request if you do not find an existing feature open.

## Identity Provider Scope

<Aside type="version">
In version 1.62.0 and later FusionAuth supports creating tenant-scoped Identity Provider instances.
</Aside>

A tenant-scoped identity provider is an instance of an identity provider that is only usable in authentication workflows for the specified tenant. This can be useful to scope a particular identity provider configuration to users within a specific tenant.

An identity provider instance that is not scoped to a specific tenant is global and available to all tenants. All identity provider instances created prior to version 1.62.0 are global.

### Global Identity Providers and Tenant-scoped API Keys

> A tenant-scoped API key cannot make changes that could be seen or otherwise affect other tenants than the one it is scoped to. Because global identity providers span all tenants, you cannot use a tenant-scoped API key with a global identity provider via the [Identity Provider API](/docs/apis/identity-providers). You can, however, use a tenant-scoped API key to interact with things like identity provider links, as these are confined to a user within a tenant.

Reviewer: ```suggestion
A tenant-scoped API key can only alter its associated tenant. Because global identity providers span all tenants, you cannot use a tenant-scoped API key with a global identity provider via the [Identity Provider API](/docs/apis/identity-providers). You can, however, use a tenant-scoped API key to interact with things like identity provider links, as these are confined to a user within a tenant.
```
[suggestion]: this first sentence is complex enough to be a bit confusing. any chance we could trim the verbosity and negatives?

### Configure an Identity Provider as Tenant-scoped

> You can only set an identity provider as tenant-scoped when it is created. Do this by setting `identityProvider.tenantId` to the desired tenant Id when using the API, or select a specific tenant from the dropdown on the Add Identity Provider view in the admin app.

Reviewer: ```suggestion
Once created, you cannot modify the scope of an identity provider. Choose a scope during identity provider creation. When using the API, set `identityProvider.tenantId` to the desired tenant ID. In the admin app, select a specific tenant from the dropdown on the Add Identity Provider view.
```
[suggestion]: reword to avoid using potentially confusing 'it' pronoun; break the complex idea in the first sentence into two simpler ideas (read-only, set during creation); split up the long second sentence into two separate sentences that begin with the context ('in the admin app' or 'when using the API')
